Sources/SKSupport/PipeAsStringHandler.swift

```diff
@@ -0,0 +1,49 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+/// Gathers data from a stdout or stderr pipe. When it has accumulated a full line, calls the handler to handle the
+/// string.
+public actor PipeAsStringHandler {
+  /// Queue on which all data from the pipe will be handled. This allows us to have a
+  /// nonisolated `handle` function but ensure that data gets processed in order.
+  private let queue = AsyncQueue<Serial>()
+  private var buffer = Data()
+
+  /// The closure that actually handles
+  private let handler: @Sendable (String) -> Void
+
+  public init(handler: @escaping @Sendable (String) -> Void) {
+    self.handler = handler
+  }
+
+  private func handleDataFromPipeImpl(_ newData: Data) {
+    self.buffer += newData
+    while let newlineIndex = self.buffer.firstIndex(of: UInt8(ascii: "\n")) {
```

Comment:
Scanning the whole buffer every time doesn't seem ideal, but I guess lines aren't that long in practice?

Reply:
Yes, the expectation is that lines aren’t long and I can’t think of common scenarios where you get less than a full line from a pipe.